data-node/src/main/java/org/graylog/datanode/configuration/OpensearchConfigurationProvider.java

```diff
@@ -120,7 +120,10 @@ private ImmutableMap<String, String> commonOpensearchConfig(final Configuration
                 networkHost -> config.put("network.host", networkHost));
         config.put("path.data", Path.of(localConfiguration.getOpensearchDataLocation()).resolve(nodeName).toAbsolutePath().toString());
         config.put("path.logs", Path.of(localConfiguration.getOpensearchLogsLocation()).resolve(nodeName).toAbsolutePath().toString());
-        config.put("cluster.initial_master_nodes", initialManagerNodes);
+
+        if(StringUtils.isNotBlank(localConfiguration.getInitialManagerNodes())) {
+            config.put("cluster.initial_master_nodes", localConfiguration.getInitialManagerNodes());
+        }
```

Comment:
I am not completely in the topic, but I just wander if we address somewhere this important advice: https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-discovery-bootstrap-cluster.html

```
After the cluster has formed, remove the cluster.initial_master_nodes setting from each node’s configuration. It should not be set for master-ineligible nodes, master-eligible nodes joining an existing cluster, or nodes which are restarting.

If you leave cluster.initial_master_nodes in place once the cluster has formed then there is a risk that a future misconfiguration may result in bootstrapping a new cluster alongside your existing cluster. It may not be possible to recover from this situation without losing data.
```

Reply:
no we don't. it is already on my internal task list but I'll put it in our board